asetypes/nullableTypes.go

```diff
@@ -0,0 +1,132 @@
+// asetypes.NullDecimal
+// asetypes.NullBinary
+// ...
+//
+// For orientation:
+// https://cs.opensource.google/go/go/+/refs/tags/go1.17.2:src/database/sql/sql.go;l=175
+
+type NullDecimal struct {
+	Dec   *Decimal
+	Valid bool
+}
+
+// Scan implements the scanner interface.
+// TODO: Discuss if replacements of sql.convertAssign-functionalities
+// should be placed in a custom asetypes.convertAssign method, see
+// https://cs.opensource.google/go/go/+/refs/tags/go1.17.2:src/database/sql/convert.go;l=210
+func (nd *NullDecimal) Scan(value interface{}) error {
+	// Replaces sql.convertAssign-functionality of checking
+	// nil-destination
+	if nd == nil {
+		return fmt.Errorf("destination pointer is nil")
+	}
+
+	if value == nil {
+		nd.Dec, nd.Valid = &Decimal{}, false
+		return nil
+	}
+
+	// Replaces sql.convertAssign-functionality of checking type
+	dec, ok := value.(*Decimal)
+	if !ok {
+		return fmt.Errorf("asetypes: %T is not of type decimal", value)
+	}
+	// Replaces sql.convertAssign-functionality of setting value
+	nd.Dec = dec
+
+	if nd.Dec.i == nil {
+		nd.Valid = false
+		return nil
+	}
+
+	nd.Valid = true
+
+	return nil
+}
+
+// Value implements the driver Valuer interface.
+func (nd NullDecimal) Value() (driver.Value, error) {
+	if !nd.Valid {
+		return nil, nil
+	}
+	return nd.Dec, nil
+}
+
+type NullBinary struct {
+	ByteSlice []byte
+	Valid     bool
+}
+
+// Scan implements the scanner interface.
+// TODO: Discuss if replacements of sql.convertAssign-functionalities
+// should be placed in a custom asetypes.convertAssign method, see
+// https://cs.opensource.google/go/go/+/refs/tags/go1.17.2:src/database/sql/convert.go;l=210
+func (nb *NullBinary) Scan(value interface{}) error {
+	// Replaces sql.convertAssign-functionality of checking
+	// nil-destination
+	if nb == nil {
+		return fmt.Errorf("destination pointer is nil")
+	}
+
+	if value == nil {
+		nb.ByteSlice, nb.Valid = []byte{}, false
+		return nil
+	}
+
+	// Replaces sql.convertAssign-functionality of checking type
+	b, ok := value.([]byte)
+	if !ok {
+		return fmt.Errorf("asetypes: %T is not of type []byte", value)
+	}
+	// Replaces sql.convertAssign-functionality of setting value
+	nb.ByteSlice = b
+	nb.Valid = true
+
+	return nil
+}
+
+// Value implements the driver Valuer interface.
+func (nb NullBinary) Value() (driver.Value, error) {
+	if !nb.Valid {
+		return nil, nil
+	}
+	return nb.ByteSlice, nil
+}
+
+var TypeToNullableTypes = map[DataType]DataType{
+	DATE:       DATEN,
+	DATETIME:   DATETIMEN,
+	SHORTDATE:  DATETIMEN,
+	FLT4:       FLTN,
+	FLT8:       FLTN,
+	INT1:       INTN,
+	INT2:       INTN,
+	INT4:       INTN,
+	INT8:       INTN,
+	MONEY:      MONEYN,
+	SHORTMONEY: MONEYN,
+	TIME:       TIMEN,
+	UINT2:      UINTN,
+	UINT4:      UINTN,
+	UINT8:      UINTN,
+}
+
+func (t DataType) NullableType() (DataType, error) {
+	nullableType, ok := TypeToNullableTypes[t]
+	if !ok {
+		return t, fmt.Errorf("nullable datatype for %s not supported", t)
```

Comment:
```suggestion
		return t, fmt.Errorf("datatype %q has no nullable equivalent", t)
```

The current message sounds like the implementation doesn't support the datatype as nullable.